### PR TITLE
[ENG-357] Document /search/ API endpoints 

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

To document the existence of our search!

## Changes

Add docs for general search endpoint: `/v2/search/` and the filtered endpoints `/v2/search/files` etc. Also adds some search syntax info.

## Side Effect

None that I know of.

## Tickets

https://openscience.atlassian.net/browse/ENG-357

### CODE REVIEW

**Comment by pattisdr** [comment]
🙌 yay docs!

File: swagger-spec/swagger.yaml [hunk 1]
```diff
@@ -35,6 +35,7 @@ x-tagGroups:
       - Entities & Entity Collections
       - External Libraries
       - Changelog
+      - Search
```
Comment (UdayVarkhedkar) [inline swagger-spec/swagger.yaml]:
```suggestion
        - Searching
```

File: swagger-spec/swagger.yaml [hunk 2]
```diff
@@ -977,6 +979,20 @@ tags:
 
       + Deprecate the `node_links` field in the **Node** and **Registration** serializers in favor of `linked_nodes` and `linked_registrations`.
 
+  - name: Search
```
Comment (UdayVarkhedkar) [inline swagger-spec/swagger.yaml]:
```suggestion
      - name: Searching
```

File: swagger-spec/search/components.yaml [hunk 3]
```diff
@@ -1 +1,159 @@
 # /search/components/
+get:
+  summary: Search the OSF for components
+  description: >-
+    This endpoint returns components that have been found by the given Elasticsearch query. All components returned
+    are serialized a components.
```
Comment (UdayVarkhedkar) [inline swagger-spec/search/components.yaml]:
```suggestion
are serialized components.
```

File: swagger-spec/search/definition.yaml [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+title: Search
+required:
+  - id
+  - type
+  - attributes
+properties:
+  data:
+    type: array
+    title: Data
+    description: "The search results, each item in this list will be serialized according to it's type."
```
Comment (UdayVarkhedkar) [inline swagger-spec/search/definition.yaml]: Missing semicolon? And additional apostrophe.
`The search results; each item in this list will be serialized according to its type.`

File: swagger-spec/search/files.yaml [hunk 5]
```diff
@@ -1 +1,86 @@
 # /search/files/
+get:
+  summary: Search the OSF for files
+  description: >-
+    This endpoint returns that have been found by the given Elasticsearch query. All files returned are serialized as
```
Comment (UdayVarkhedkar) [inline swagger-spec/search/files.yaml]: Missing "files"

File: swagger-spec/search/projects.yaml [hunk 6]
```diff
@@ -1 +1,159 @@
 # /search/projects/
+get:
+  summary: Search the OSF for projects
+  description: >-
+    This endpoint returns projects that have been found by the given Elasticsearch query. All projects returned
+    are serialized a projects.
```
Comment (UdayVarkhedkar) [inline swagger-spec/search/projects.yaml]: Same typo - Either 'are serialized as projects' or 'are serialized projects' or 'are serialized' should be fine.

File: swagger-spec/search/registrations.yaml [hunk 7]
```diff
@@ -1 +1,159 @@
 # /search/registrations/
+get:
+  summary: Search the OSF for registrations
+  description: >-
+    This endpoint returns registrations that have been found by the given Elasticsearch query. All registrations
+    returned are serialized a registrations.
```
Comment (UdayVarkhedkar) [inline swagger-spec/search/registrations.yaml]: Same typo - Either 'are serialized as registrations' or 'are serialized registrations' or 'are serialized' should be fine.

File: swagger-spec/search/users.yaml [hunk 8]
```diff
@@ -1 +1,68 @@
 # /search/users/
+get:
+  summary: Search the OSF for users
+  description: >-
+    This endpoint returns users that have been found by the given Elasticsearch query. All users returned
+    are serialized a user.
```
Comment (UdayVarkhedkar) [inline swagger-spec/search/users.yaml]: Same typo - Either 'are serialized as users' or 'are serialized users' or 'are serialized' should be fine.

File: swagger-spec/search/search.yaml [hunk 9]
```diff
@@ -1 +1,26 @@
 # /search/
+get:
+  summary: Search the OSF
+  description: >-
+    Objects (including projects, components, registrations, users, files, and institutions) that have been found by the
```
Comment (UdayVarkhedkar) [inline swagger-spec/search/search.yaml]: Perhaps should start with "Returns objects (including projects, ..."

**Pull request merged**